nullaway/build.gradle

```diff
@@ -32,6 +32,8 @@ dependencies {
     compileOnly deps.build.errorProneCheckApi
     implementation deps.build.checkerDataflow
     implementation deps.build.guava
+    // for InferredJARModelsHandler classpath scanning
+    implementation deps.build.classgraph
```

Comment:
As fas I can tell, using ClassGraph is the sanest way to do classpath scanning after Java 9.  However, I am very hesitant to add another required dependence for NullAway.  So what I'm thinking is, we could add this dependence as `compileOnly`, and then any client who is using JarInfer (probably quite a small percentage of our users) would also have to add ClassGraph to their processing path.  We'd have to document this need, of course.  We could use reflection to give a good error message in the case where reading JarInfer annotations is enabled but the ClassGraph library is not present.

@lazaroclapp two questions:
1. What do you think of the proposal above?
2. Would the ClassGraph dependence cause any problems at Uber, in cases where JarInfer is being used?

Reply:
While I don't like adding more dependencies to NullAway, I am also worried about having more required build configuration fiddling than we need to, when enabling NullAway options. The idea of "if you pass this flag, you also need to add this other library dep" is not something that sits entirely well with me, but of course might be the lesser of two evils if:

- This `ClassGraph` library is large, messy, or has a large number of dependencies itself
- There is no JDK 11 native way of doing this scan
- We still need to do this scan (see larger comment about the three JarInfer modes)